### PR TITLE
ch4: More 32-bit fixes

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Another handful of patches to fix warnings in 32-bit builds.

## Expected Performance Changes

none

## Known Issues

none

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [x] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [x] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

File: src/mpid/ch4/netmod/ofi/util.c [hunk 1]
```diff
@@ -501,9 +501,11 @@ static inline void create_dt_map()
 
 static inline void add_index(MPI_Datatype datatype, int *idx)
 {
-    MPIR_Datatype *dt_ptr;
-    MPIR_Datatype_get_ptr(datatype, dt_ptr);
-    MPIDI_OFI_DATATYPE(dt_ptr).index = *idx;
+    if (datatype != MPI_DATATYPE_NULL) {
```
Comment (hajimefu) [inline src/mpid/ch4/netmod/ofi/util.c]: I would add a comment here that some datatypes may not be supported depending on the underlying platform, in which case `datatype` becomes `MPI_DATATYPE_NULL`.

Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/util.c]: OK, will add.

**Comment by raffenet** [comment]
One last sanity check.
test:jenkins/ch4/ofi

**Pull request merged**